packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/ios/enrollmentintegrationsteps.md

```diff
@@ -0,0 +1,523 @@
+                        Text(name)
+                    }
+                    Image(uiImage: qrCode)
+                        .resizable()
+                        .scaledToFit()
+                        .frame(width: 200, height: 200)
+                    Text("Enroll by scanning this code in the app or use the following code:")
+                    Text((enrollData?.sharedSecret)!)
+                }
+            }
+            // Display only if there's a message.
+            if let message = message {
+                Section("Message") {
+                    HStack {
+                        Image(systemName: "exclamationmark.triangle.fill")
+                            .foregroundColor(Color.yellow)
+                        Text(message)
+                    }
+                }
+            }
+            Section("Code") {
+                let textLabel = (codeData?.entryField.label != nil) ?
+                codeData?.entryField.label : "Enter the code"
+                TextField(textLabel!,text: $code)
+                    .textInputAutocapitalization(.never)
+                    .disableAutocorrection(true)
+                    .keyboardType(.numberPad)
+            }
+            Section() {
+                Button("Continue") {
+                    continueSignin()
+                }
+                Button("Cancel Sign In", role: .destructive) {
+                    showingSignIn = false
+                }
+            }
+
+        }
+        .onAppear() {
+            if let signInController = SignInController.shared {
+                enrollData = signInController.authenticatorEnrollment()
+                codeData = signInController.authenticatorCodeFields()
+            }
+        }
+    }
+
+    func skip() {
+        print("SKIP")
+
+    }
+
+    func continueSignin() {
+        print("Continue")
+        SignInController.shared?.authenticateWithCode(code: code) {result in
+             isWaiting = false
+            switch result {
+            case .success(let result):
+                isWaiting = false
+                    // Succesful sign in
+                    signedIn = true
+                    showingSignIn = false
+                } else if let message = result.message {
+                    // Show a message on this form.
+                    self.message = message.message
+                } else {
+                    currentState = SignInController.shared!.currentRemediation!.type
+                }
+
+            case .failure(let error):
+                // Handle the error
+            }
+        }
+    }
+}
+```
+
+There are three possible results after sending the code to the server:
+
+- Sending the correct code signs the user in.
+- Sending an incorrect code results in another `.enrollAuthenticator` response with a message indicating that the code doesn't match. The UI code handles that case by displaying a form that includes the message.
+- An error, such as a disconnected server.
+
+
```

Comment:
It feels like this just stops here. 
Perhaps show how to retrieve the User's OIDC claims from the server when the user has successfully signed in?

Reply:
I disagree about this being up to the skeleton of the article as a whole. 
The end of the flow is having the user be successfuly authenticated so if you were going to do something in the article post-successful login, it would be in the stack selector. 

Pulling in the claims is only three lines of code in C# so I've included them. If it's a lot more in Swift, maybe it's easier to exclude it.

Reply:
Good point... it's missing the final step of requesting a token and dealing with the callback containing that token. I added this:

A successful sign-on sets `isLoginSuccessful` to `true`. Your code then aquires a token from the response by calling `response.exchangeCode()`. The following code shows the singleton delegate function called that's called with the token:

```swift
extension SignInController: IDXClientDelegate {
    // Delegate method sent after a token is exchanged.
    public func idx(client: IDXClient, didReceive token: Token) {
        // Inform the UI that sign-on is successful
    }
    ...
}
```

Reply:
sp.
Your code then ACQUIRES a token

Reply:
Nice catch. I also modified as acquires seems like an odd choice of word. This change will be in the next commit:

> A successful sign-on sets `isLoginSuccessful` to `true`. Your code then requests a token from the server by calling `response.exchangeCode()`. The following code shows the singleton delegate function called that's called with the returned token: